library/query_ref_context_injection.yml

```diff
@@ -0,0 +1,30 @@
+id: RP-9
+
+info:
+  name: Branch Context Injection
+  severity: high
+  description: Branch Injection is caused by using ref/default_branch variables in inline scripts
+  full-description: |
+    Creating a new pull request could be submitted with a branch name that contains special characters.
+    By using a ref/default_branch variable in an inline script, an attacker can inject arbitrary code into the build process.
+  references:
+    - https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions
```

Comment:
https://cycode.com/blog/ci-story-how-we-found-critical-vulnerabilities-in-storybook-project/